File: openfeature-provider/java/src/main/java/com/spotify/confidence/sdk/ResolverApi.java
package com.spotify.confidence.sdk;

import com.spotify.confidence.sdk.flags.resolver.v1.ResolveFlagsResponse;
import com.spotify.confidence.sdk.flags.resolver.v1.ResolveProcessRequest;
import java.util.concurrent.CompletionStage;

/** Common interface for WASM-based flag resolver implementations. */
interface ResolverApi {

  /**
   * Resolves flags with materialization support (suspend/resume).
   *
   * @param request The resolve process request
   * @return A future containing the resolve response
   */
  CompletionStage<ResolveFlagsResponse> resolveProcess(ResolveProcessRequest request);

  void init(byte[] state, String accountId);

  /** Returns true if the resolver has been initialized with valid state. */
  boolean isInitialized();

  /**
   * Updates the resolver state and flushes any pending logs.
   *
   * @param state The new resolver state
   * @param accountId The account ID
   */
  void updateStateAndFlushLogs(byte[] state, String accountId);

  /** Closes the resolver and releases any resources. */
  void close();
}

File: openfeature-provider/android/confidence-provider/src/main/kotlin/com/spotify/confidence/android/ResolverApi.kt
package com.spotify.confidence.android

import com.spotify.confidence.flags.resolver.v1.ResolveFlagsResponse
import com.spotify.confidence.flags.resolver.v1.ResolveWithStickyRequest
import java.util.concurrent.CompletableFuture

/**
 * Interface for the resolver API that handles flag resolution.
 */
internal interface ResolverApi {
    /**
     * Initializes the resolver with the given state and account ID.
     */
    fun init(state: ByteArray, accountId: String)

    /**
     * Returns whether the resolver has been initialized.
     */
    fun isInitialized(): Boolean

    /**
     * Updates the resolver state and flushes any pending logs.
     */
    fun updateStateAndFlushLogs(state: ByteArray, accountId: String)

    /**
     * Resolves flags with sticky assignment support.
     */
    fun resolveWithSticky(request: ResolveWithStickyRequest): CompletableFuture<ResolveFlagsResponse>

    /**
     * Closes the resolver, flushing any pending logs.
     */
    fun close()
}
